clients/src/main/java/org/apache/kafka/clients/admin/ConfigEntry.java

```diff
@@ -25,34 +35,66 @@
     private final boolean isSensitive;
     private final boolean isReadOnly;
 
+    /**
+     * Create a configuration entry with the provided values.
+     *
+     * @param name the non-null config name
+     * @param value the config value or null
+     */
     public ConfigEntry(String name, String value) {
         this(name, value, false, false, false);
     }
 
+    /**
+     * Create a configuration with the provided values.
+     *
+     * @param name the non-null config name
+     * @param value the config value or null
+     * @param isDefault whether the config value is the default or if it's been explicitly set
+     * @param isSensitive whether the config value is sensitive, the broker never returns the value if it is sensitive
+     * @param isReadOnly whether the config is read-only and cannot be updated
+     */
     public ConfigEntry(String name, String value, boolean isDefault, boolean isSensitive, boolean isReadOnly) {
+        Objects.requireNonNull(name, "name should not be null");
         this.name = name;
         this.value = value;
         this.isDefault = isDefault;
         this.isSensitive = isSensitive;
         this.isReadOnly = isReadOnly;
     }
 
+    /**
+     * Return the config name.
+     */
     public String name() {
         return name;
     }
 
+    /**
+     * Return the value or null. Null is returned if the config is unset or if isSensitive is true.
+     */
     public String value() {
         return value;
     }
 
+    /**
+     * Return whether the config value is the default or if it's been explicitly set.
+     */
     public boolean isDefault() {
         return isDefault;
     }
 
+    /**
+     * Return whether the config value is sensitive. The value is always set to null by the broker if the config value
+     * is sensitive.
+     */
     public boolean isSensitive() {
```

Comment:
Kafka drops getters, but booleans are a little difference in that they can be pretty unclear without a prefix. I noticed this and a few other inconsistencies, but wanted to keep this PR purely as Javadoc changes.